R/CreatePackageVignette.R

```diff
@@ -0,0 +1,101 @@
+#' @title pkgnet Report as Vignette
+#' @name CreatePackageVignette
+#' @description Create pkgnet package report as an Rmarkdown vignette. This
+#'    vignette is able to be rendered with the
+#'    \code{\link[knitr:vignette_engines]{knitr::rmarkdown}} vignette engine
+#'    into HTML vignettes upon package building. It is also compatible with
+#'    \code{\link[pkgdown:build_articles]{pkgdown}} sites.
+#' @param pkg_name (string) name of a package
+#' @param pkg_path (string) The path to the package repository. If given, coverage
+#'                 will be calculated for each function. \code{pkg_path} can be an
+#'                 absolute or relative path.
+#' @param pkg_reporters (list) a list of initialized package reporters
+#' @param vignette_path (string) The path and filename of the output vignette
+#'    file. The default assumes your working directory is the package root.
+#' @importFrom rlang enexpr
+#' @importFrom assertthat assert_that is.string is.readable
+#' @importFrom tools file_ext
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
+                                  , pkg_path = NULL
+                                  , pkg_reporters = list(
+                                      DependencyReporter$new()
+                                      , FunctionReporter$new()
+                                  )
+                                  , vignette_path = file.path("vignettes"
+                                                              , "pkgnet_report.Rmd")
+                                 ) {
+
+    # Capture pkg_reporters expression for later injection into Rmd
+    pkg_reporters_expr <- rlang::enexpr(pkg_reporters)
+
+    ## pkg_name input checks ##
+    assertthat::assert_that(
+        assertthat::is.string(pkg_name)
+        , pkg_name != ""
+    )
+
+    ## pkg_path input checks ##
+    assertthat::assert_that(
+        is.null(pkg_path) || assertthat::is.readable(pkg_path)
+    )
+
+    ## vignette_path input checks ##
+    assertthat::assert_that(
+        assertthat::is.string(vignette_path)
+        , vignette_path != ""
+        , identical(tolower(tools::file_ext(vignette_path)), "rmd")
+    )
+    # Confirm directory exists
+    if (!assertthat::is.dir(dirname(vignette_path))) {
+        log_fatal(sprintf("directory %s does not exist, please create first"))
+    }
+
+    ## pkg_reporter input checks ##
+    assertthat::assert_that(
+        is.list(pkg_reporters)
+    )
+    # Check if generators were passed in by accident
+    if (any(vapply(pkg_reporters, FUN = R6::is.R6Class, FUN.VALUE = logical(1)))) {
```

Comment:
why `logical(1)` and not `FALSE`?

Reply:
I think using `FALSE` would do the same thing right? I think `logical(1)` is easier to read -- it must be a logical vector of length 1. 

From `vapply`'s doc
```
This function checks that all values of FUN are compatible with the FUN.VALUE, in that they must have the same length and type.
```

Reply:
I see what you're getting at here.